src/plugins/es_ui_shared/public/request/use_request.ts

```diff
@@ -65,49 +65,53 @@ export const useRequest = <D = any, E = Error>(
-    const { data: serializedResponseData, error: responseError } = response;
-
-    const isOutdatedRequest = requestId !== requestCountRef.current;
-    const isUnmounted = isMounted.current === false;
-
-    // Ignore outdated or irrelevant data.
-    if (isOutdatedRequest || isUnmounted) {
-      return;
-    }
-
-    // Surface to consumers that at least one request has resolved.
-    isInitialRequestRef.current = false;
+  const resendRequest = useCallback(
+    async (asSystemRequest?: boolean) => {
+      // If we're on an interval, this allows us to reset it if the user has manually requested the
+      // data, to avoid doubled-up requests.
+      clearPollInterval();
 
-    setError(responseError);
-    // If there's an error, keep the data from the last request in case it's still useful to the user.
-    if (!responseError) {
-      const responseData = deserializer
-        ? deserializer(serializedResponseData)
-        : serializedResponseData;
-      setData(responseData);
-    }
-    // Setting isLoading to false also acts as a signal for scheduling the next poll request.
-    setIsLoading(false);
-  }, [requestBody, httpClient, deserializer, clearPollInterval]);
+      const requestId = ++requestCountRef.current;
+
+      // We don't clear error or data, so it's up to the consumer to decide whether to display the
+      // "old" error/data or loading state when a new request is in-flight.
+      setIsLoading(true);
+
+      const requestPayload = { ...requestBody, asSystemRequest };
+      const response = await sendRequest<D, E>(httpClient, requestPayload);
+      const { data: serializedResponseData, error: responseError } = response;
+
+      const isOutdatedRequest = requestId !== requestCountRef.current;
+      const isUnmounted = isMounted.current === false;
+
+      // Ignore outdated or irrelevant data.
+      if (isOutdatedRequest || isUnmounted) {
+        return;
+      }
+
+      // Surface to consumers that at least one request has resolved.
+      isInitialRequestRef.current = false;
+
+      setError(responseError);
+      // If there's an error, keep the data from the last request in case it's still useful to the user.
+      if (!responseError) {
+        const responseData = deserializer
+          ? deserializer(serializedResponseData)
+          : serializedResponseData;
+        setData(responseData);
+      }
+      // Setting isLoading to false also acts as a signal for scheduling the next poll request.
+      setIsLoading(false);
+    },
+    [requestBody, httpClient, deserializer, clearPollInterval]
+  );
 
   const scheduleRequest = useCallback(() => {
     // If there's a scheduled poll request, this new one will supersede it.
     clearPollInterval();
 
     if (pollIntervalMs) {
-      pollIntervalIdRef.current = setTimeout(resendRequest, pollIntervalMs);
+      pollIntervalIdRef.current = setTimeout(() => resendRequest(true), pollIntervalMs);
```

Comment:
I am not sure I follow. The consumer can decide to pass it to the config (`SendRequestConfig` in "send_request.ts") but here we force it to `true`? Do you mind explaining? Do we always want it to be `true`?

Reply:
As discussed in https://github.com/elastic/kibana/pull/91070/commits/8006eb7b289dd0789430982dd5ce095f1d942abd#r576262009, if this request is sent during background polling, it should be treated as a system request. Consumers should not be able to change that behavior.

Reply:
@jportner Sorry, I don't see a discussion at the link you shared. Could you help me understand your line of reasoning? What's the relationship between a poll and a system request? FWIW, the "poll" in this case simply defines behavior -- the request is resent automatically on an interval. This logic is otherwise unconcerned about the consumer's use case.

Reply:
Sorry, clipboard must have gotten screwed up. I edited the link to fix it. I was referring to my other comment above.

Let me rephrase: if a request can be sent (or resent) in the background, _without user interaction_, it should be flagged as a system request. Polling-style behavior such as this fits the bill.

Reply:
Thanks Joe! I just read the linked issue in the PR description, which I should have done in the first place. I think I understand what you're trying to do now. We want to detect when a session has become idle. We're using requests to the server as a proxy for user interaction. Requests that are sent on a poll are incorrectly interpreted as user interaction, preventing idle sessions from being detected. `asSystemRequest` enables the server to ignore these and similar requests, as a solution the problem.

I think this approach makes sense. My only concern is that the code doesn't communicate this intention, and possibly signals the wrong intentions because of the term "system". We have system indices in Elasticsearch, which carry connotations about restricted access and usage. In this case, "system" feels like too broad and powerful a term for what is a much narrower and less consequential scope. Can we find a name that matches this narrower scope? For example, `asIdleRequest`?

Reply:
> ...
> `asSystemRequest` enables the server to ignore these and similar requests, as a solution the problem.

Yes!

> I think this approach makes sense.
> ...
> Can we find a name that matches this narrower scope? For example, `asIdleRequest`?

The `kbn-system-request` header has been around for a long time. I agree that calling this a "system request" is a terrible name, I didn't (and wouldn't) choose it 😄 

So, the http plugin exposes its functions with the `HttpFetchOptions` parameter, that has the `asSystemRequest` field and sets the `kbn-system-request` header accordingly.

We could change the parameter name to something else in these `useRequest` and `sendRequest` functions, but I am sort of concerned that would cause even more confusion that something like `asIdleRequest` would control this header. Especially if/when we are trying to find all usages of the `kbn-system-request` header in the future, this would make that more difficult.

How would you feel if I leave the name as-is, and add some comments in these two functions to explain the rationale, and add a TSdoc in the `SendRequestConfig` interface?

Reply:
> concerned that would cause even more confusion

Good point I agree.

> How would you feel if I leave the name as-is, and add some comments in these two functions to explain the rationale, and add a TSdoc in the SendRequestConfig interface?

This sounds like a great solution. Thanks for offering it up. :)

> added some comments

I think these are very helpful! Thank you so much for hearing me out, even though I should have spent more time familiarizing myself with the PR before commenting. 😅